tools/sapp/sapp/interactive.py

```diff
@@ -1648,17 +1673,22 @@ def _get_leaves_trace_frame(
     def _leaf_dict_lookups(
         self, message_ids: List[int], kind: SharedTextKind
     ) -> Set[str]:
-        leaf_dict = (
-            self.sources_dict if kind == SharedTextKind.SOURCE else self.sinks_dict
-        )
+        if kind == SharedTextKind.SOURCE:
+            leaf_dict = self.sources_dict
+        elif kind == SharedTextKind.SINK:
+            leaf_dict = self.sinks_dict
+        else:
```

Comment:
I have a minor preference for explicitly `elif`ing on `SharedTextKind.FEATURE` here and raising in the `else`, just as a sanity check. Feel free to ignore though, since you're already following the previously existing pattern of using the `else` to assume the expected `SharedTextKind`

Reply:
Looking at the use of `raise` on the code, it's more tied to the `UserError` (used when a user make a mistake with their command). It looks like user don't have the ability to directly call this function and this mostly will be used internally inside the code. We can just use `elfi` and don't have an `else` or use switch here but I think error with stacktrace would be more helpful for future development instead of raise.